src/python/pants/goal/run_tracker.py

```diff
@@ -428,6 +433,30 @@ def end_workunit(self, workunit):
       self.self_timings.add_timing(path, self_time, is_tool)
       self.outcomes[path] = workunit.outcome_string(workunit.outcome())
 
+  def get_critical_path_timings(self):
+    """
+    Get the cumulative timings of each goal and all of the goals it (transitively) depended on.
+    """
+    transitive_dependencies = dict()
+    for goal_info in self._sorted_goal_infos:
+      deps = transitive_dependencies.setdefault(goal_info.goal.name, set())
+      for dep in goal_info.goal_dependencies:
+        deps.add(dep.name)
+        deps.update(transitive_dependencies.get(dep.name))
+
+    raw_timings = dict()
+    for entry in self.cumulative_timings.get_all():
+      raw_timings[entry["label"]] = entry["timing"]
+
+    timings = AggregatedTimings()
+    for goal, deps in transitive_dependencies.items():
+      label = "main:{}".format(goal)
```

Comment:
Perhaps overkill, but: there is a constant for the `DEFAULT_ROOT_NAME` that you could use here.